lib/ts/recipe/dashboard/api/multitenancy/getThirdPartyConfig.ts

```diff
@@ -0,0 +1,254 @@
+/* Copyright (c) 2024, VRAI Labs and/or its affiliates. All rights reserved.
+ *
+ * This software is licensed under the Apache License, Version 2.0 (the
+ * "License") as published by the Apache Software Foundation.
+ *
+ * You may not use this file except in compliance with the License. You may
+ * obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+import { APIInterface, APIOptions } from "../../types";
+import Multitenancy from "../../../multitenancy";
+import MultitenancyRecipe from "../../../multitenancy/recipe";
+import {
+    findAndCreateProviderInstance,
+    mergeProvidersFromCoreAndStatic,
+} from "../../../thirdparty/providers/configUtils";
+import { ProviderConfig } from "../../../thirdparty/types";
+import { UserContext } from "../../../../types";
+
+export type Response =
+    | {
+          status: "OK";
+          providerConfig: ProviderConfig & {
+              isGetAuthorisationRedirectUrlOverridden: boolean;
+              isExchangeAuthCodeForOAuthTokensOverridden: boolean;
+              isGetUserInfoOverridden: boolean;
+          };
+      }
+    | {
+          status: "UNKNOWN_TENANT_ERROR";
+      };
+
+export default async function getThirdPartyConfig(
+    _: APIInterface,
+    tenantId: string,
+    options: APIOptions,
+    userContext: UserContext
+): Promise<Response> {
+    let tenantRes = await Multitenancy.getTenant(tenantId, userContext);
+
+    if (tenantRes === undefined) {
+        return {
+            status: "UNKNOWN_TENANT_ERROR",
+        };
+    }
+
+    const thirdPartyId = options.req.getKeyValueFromQuery("thirdPartyId");
+
+    if (thirdPartyId === undefined) {
+        throw new Error("Please provide thirdPartyId");
+    }
+
+    let providersFromCore = tenantRes?.thirdParty?.providers;
+    const mtRecipe = MultitenancyRecipe.getInstance();
+    const staticProviders = mtRecipe?.staticThirdPartyProviders ?? [];
+
+    if (providersFromCore === undefined) {
+        providersFromCore = [];
+    }
+
+    let found = false;
+    for (let i = 0; i < providersFromCore.length; i++) {
+        if (providersFromCore[i].thirdPartyId === thirdPartyId) {
+            found = true;
+            if (thirdPartyId === "okta") {
+                providersFromCore[i].oidcDiscoveryEndpoint = options.req.getKeyValueFromQuery("oktaDomain");
```

Comment:
add comments here explaining what is happening

Reply:
Done